src/go/rpk/pkg/config/params.go

```diff
@@ -346,9 +358,122 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 		return nil, err
 	}
 	c.addUnsetDefaults()
+
+	if _, err := p.Logger(); err != nil { // ensure our logger is loaded so MustLogger can be used
+		return nil, err
+	}
 	return c, nil
 }
 
+// LoadedLogger is like Logger, but panics on error. This can be used if you
+// are sure Load has been called, which internally sets the logger and returns
+// the logger error, if any.
+func (p *Params) LoadedLogger() *zap.Logger {
+	logger, err := p.Logger()
+	if err != nil {
+		panic(err)
+	}
+	return logger
+}
+
+// SugarLogger returns LoadedLogger().Sugar().
+func (p *Params) SugarLogger() *zap.SugaredLogger {
+	return p.LoadedLogger().Sugar()
+}
+
+// Logger parses p.LogLevel and returns the corresponding zap logger or an
+// error.
+func (p *Params) Logger() (*zap.Logger, error) {
+	p.loggerOnce.Do(func() {
+		// First we normalize the level. We support prefixes such
+		// that "w" means warn.
+		p.LogLevel = strings.TrimSpace(strings.ToLower(p.LogLevel))
+		if p.LogLevel == "" {
+			p.LogLevel = "none"
+		}
+		var ok bool
+		for _, level := range []string{"none", "error", "warn", "info", "debug"} {
+			if strings.HasPrefix(level, p.LogLevel) {
+				p.LogLevel, ok = level, true
+				break
+			}
+		}
+		if !ok {
+			p.loggerErr = fmt.Errorf("invalid log level: %s", p.LogLevel)
+			return
+		}
+		var level zapcore.Level
+		switch p.LogLevel {
+		case "none":
+			p.logger = zap.NewNop()
+			return
+		case "error":
+			level = zap.ErrorLevel
+		case "warn":
+			level = zap.WarnLevel
+		case "info":
+			level = zap.InfoLevel
+		case "debug":
+			level = zap.DebugLevel
+		}
+
+		// Now the zap config. We want to to the console and make the logs
+		// somewhat nice. The log time is effectively time.TimeMillisOnly.
+		// We disable logging the callsite and sampling, we shorten the log
+		// level to three letters, and we only add color if this is a
+		// terminal.
+		zcfg := zap.NewProductionConfig()
+		zcfg.Level = zap.NewAtomicLevelAt(level)
+		zcfg.DisableCaller = true
+		zcfg.DisableStacktrace = true
+		zcfg.Sampling = nil
+		zcfg.Encoding = "console"
+		zcfg.EncoderConfig.EncodeTime = zapcore.TimeEncoder(func(t time.Time, pae zapcore.PrimitiveArrayEncoder) {
+			pae.AppendString(t.Format("15:04:05.000"))
+		})
```

Comment:
[question] will this print the date?

Reply:
we never printed the date previously, and IMO printing the date gets a pretty noisy